aws/resource_aws_rds_cluster_endpoint_test.go

```diff
@@ -0,0 +1,104 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccAWSRDSClusterEndpoint_importBasic(t *testing.T) {
+	ri := acctest.RandInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSClusterEndpointConfig(ri),
+			},
+
+			{
+				ResourceName:      "aws_rds_cluster_endpoint.reader",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+
+			{
+				ResourceName:      "aws_rds_cluster_endpoint.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSRDSClusterEndpoint_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	readerResourceName := "aws_rds_cluster_endpoint.reader"
+	defaultResourceName := "aws_rds_cluster_endpoint.default"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSClusterDestroy,
```

Comment:
This resource should implement its own `CheckDestroy` function that verifies the resource has been completely removed from AWS.

Reply:
I implement own CheckDestroy function